robowflex_library/src/builder.cpp

```diff
@@ -105,10 +108,50 @@ void MotionRequestBuilder::setGoalRegion(const std::string &ee_name, const std::
     constraints.orientation_constraints.push_back(
         TF::getOrientationConstraint(ee_name, base_name, orientation, tolerances));
 
-    request_.goal_constraints.clear();
     request_.goal_constraints.push_back(constraints);
 }
 
+void MotionRequestBuilder::addGoalRotaryTile(const std::string &ee_name, const std::string &base_name,
+                                             const Eigen::Affine3d &pose, const GeometryConstPtr &geometry,
+                                             const Eigen::Quaterniond &orientation,
+                                             const Eigen::Vector3d &tolerances, const Eigen::Affine3d &offset,
+                                             const Eigen::Vector3d &axis, unsigned int n)
+{
+    double pi2 = 2 * boost::math::constants::pi<double>();
+
+    for (double angle = 0; angle < pi2; angle += pi2 / n)
+    {
+        Eigen::Quaterniond rotation(Eigen::AngleAxisd(angle, axis));
+        Eigen::Affine3d newPose = pose * rotation * offset;
+        Eigen::Quaterniond newOrientation(rotation * orientation);
+
+        setGoalRegion(ee_name, base_name, newPose, geometry, newOrientation, tolerances);
+    }
+}
+
+void MotionRequestBuilder::addCylinderSideGrasp(const std::string &ee_name, const std::string &base_name,
+                                                const Eigen::Affine3d &pose, const GeometryConstPtr &cylinder,
+                                                double distance, double depth, unsigned int n)
+{
+    // Grasping region to tile
+    auto box = Geometry::makeBox(depth, depth, cylinder->getDimensions()[1]);
+    Eigen::Affine3d offset(Eigen::Translation3d(cylinder->getDimensions()[0] + distance, 0, 0));
+
+    Eigen::Quaterniond orientation =
+        Eigen::AngleAxisd(-boost::math::constants::pi<double>(), Eigen::Vector3d::UnitX())  //
```

Comment:
I know we already use boost, but if `std` has a pi constant can we switch to it?

Reply:
Unfortunately no, the `boost::math::constants` is about as good as we can get. We could typedef it ourselves to hide the length declaration, but that's about it.

Reply:
Wow, come on c++.  Boost is fine then.